askama/src/filters/alloc.rs

```diff
@@ -475,32 +655,37 @@ pub fn capitalize(s: impl fmt::Display) -> Result<String, fmt::Error> {
 /// );
 /// # }
 /// ```
-pub fn title(s: impl fmt::Display) -> Result<String, fmt::Error> {
-    let mut buffer;
-    let s = try_to_str!(s => buffer);
-    let mut need_capitalization = true;
-
-    // Sadly enough, we can't mutate a string when iterating over its chars, likely because it could
-    // change the size of a char, "breaking" the char indices.
-    let mut output = String::with_capacity(s.len());
-    for c in s.chars() {
-        if c.is_whitespace() {
-            output.push(c);
-            need_capitalization = true;
-        } else if need_capitalization {
-            match c.is_uppercase() {
-                true => output.push(c),
-                false => output.extend(c.to_uppercase()),
-            }
-            need_capitalization = false;
-        } else {
-            match c.is_lowercase() {
-                true => output.push(c),
-                false => output.extend(c.to_lowercase()),
-            }
-        }
+#[inline]
+pub fn title<S: fmt::Display>(source: S) -> Result<Title<S>, Infallible> {
+    Ok(Title(source))
+}
+
+pub struct Title<S>(S);
+
+impl<S: fmt::Display> fmt::Display for Title<S> {
+    fn fmt(&self, dest: &mut fmt::Formatter<'_>) -> fmt::Result {
+        let mut buffer;
+        flush_title(dest, try_to_str!(self.0 => buffer))
+    }
+}
+
+impl<S: FastWritable> FastWritable for Title<S> {
+    fn write_into<W: fmt::Write + ?Sized>(
+        &self,
+        dest: &mut W,
+        values: &dyn crate::Values,
+    ) -> crate::Result<()> {
+        let mut buffer = String::new();
+        self.0.write_into(&mut buffer, values)?;
+        Ok(flush_title(dest, &buffer)?)
     }
-    Ok(output)
+}
+
+fn flush_title(dest: &mut (impl fmt::Write + ?Sized), s: &str) -> fmt::Result {
+    for word in s.split_inclusive(' ') {
```

Comment:
`' '` is a whitespace, but so is `'\n'` or `'\t'` and a few others. Take a look at the [is_whitespace](https://doc.rust-lang.org/nightly/std/primitive.char.html#method.is_whitespace) docs. 😉

Reply:
Oops, you're right. Changed to `char::is_whitespace`.